src/policy/fees.cpp

```diff
@@ -658,31 +658,97 @@ CFeeRate CBlockPolicyEstimator::estimateRawFee(int confTarget, double successThr
     return CFeeRate(median);
 }
 
-CFeeRate CBlockPolicyEstimator::estimateSmartFee(int confTarget, int *answerFoundAtTarget, const CTxMemPool& pool) const
+
+double CBlockPolicyEstimator::estimateCombinedFee(unsigned int confTarget, double successThreshold, bool conservative) const
+{
+    double estimate = -1;
```

Comment:
I can take a pass at rewriting the logic to be a bit clearer, but I'm not sure a loop is the right thing.  There are basically 6 different combinations of estimates you might want to check depending on what range your target is in and whether or not you want to check shorter horizons.  But the logic might be cleaner if those are just spelled out.